# The Tenebrio Genome project
The yellow mealworm beetle, <i>Tenebrio molitor</i>, is a promising alternative protein source for animal and human nutrition. Here we provide the methods and data to construct in 280Mb mealworm genome in a chrooosme-scale assembly now available at https://www.ebi.ac.uk/ena/browser/view/CAJRHG010000000

## Underlying and extended data
In this repository, you will find all the detailed bioinformatic wolkflows and the methods versions used for the mealworm genome assembly and genome annotation.

## Tenebrio genome assembly
For more details on the genome assembly method, you can read the notebook assembly_workflow.pdf. Some of th scripts available from the Genoscope github https://github.com/institut-de-genomique, especially the genome polisher Hapo-G (https://github.com/institut-de-genomique/HAPO-G) from JM Aury et al. Hapo-G, haplotype-aware polishing of genome assemblies with accurate reads, NAR Genomics and Bioinformatics, Volume 3, Issue 2, June 2021, lqab034, https://doi.org/10.1093/nargab/lqab034

## Tenebrio genome annotation
For more details on the genome annotation, you can read the notebook annotation_workflow.html. The genome annotation is based on the GMOVE annotater available at https://github.com/institut-de-genomique/Gmove from Dubarry et al., Gmove a tool for eukaryotic gene predictions using various evidences [version 1; not peer reviewed]. F1000Research 2016, 5(ISCB Comm J):681 (poster) (https://doi.org/10.7490/f1000research.1111735.1) 

## This work is part of the Tenebrio genomic project H2020 FARMYNG
https://www.farmyng.eu/ 

## Licenses
All codes and scripts in the annotation and assembly folder are under the GPL-3.0 license. Other underlying data are under the CC0 LICENSE.



###
For more details on the project, scripts and data please contact us at amine.madoui@gmail.om


